### PR TITLE
Graph Widget Zoom Tweaks:

### PR DESCRIPTION
**Pull request by GordonSmith**
Add optional transitions.
Allow smaller graphs.
Add more getBounds options.
Zoom to fit is now 100% (previously 90%).
Prevent auto zoom after layout.

Signed-off-by: Gordon Smith <gordon.smith@lexisnexis.com>

Conflicts:
	widgets/src/graph/Graph.js

### CODE REVIEW

**Comment by GordonSmith** [comment]
@drealeed Please Review

File: widgets/src/graph/Graph.js [hunk 1]
```diff
@@ -111,19 +110,25 @@
...
             this.zoom.scale(scale);
-            this.applyZoom();
+            this.applyZoom(transitionDuration);
         }
     };
 
-    Graph.prototype.applyZoom = function (transition) {
-        (transition ? this.svg.transition() : this.svg)
+    Graph.prototype.applyZoom = function (transitionDuration) {
+        if (d3.event && d3.event.sourceEvent && (!d3.event.sourceEvent.ctrlKey && d3.event.sourceEvent.type == 'mousewheel' || d3.event.sourceEvent.type == 'DOMMouseScroll')) {
```
Comment (drealeed) [inline widgets/src/graph/Graph.js]: Are we currently supporting tablet-based scrolling with the above two sourceevent type checks? Would it be better to check the WheelEvent interface?

Comment (drealeed) [inline widgets/src/graph/Graph.js]: Are we currently supporting tablet-based scrolling with the above two sourceevent type checks? Would it be better to check the WheelEvent interface?

**Comment by GordonSmith** [comment]
Rebased for merge.  Also tidied some of the wheel code.

FYI This works as expected on both touch and mouse devices (somewhat surprisingly!)

**Comment by drealeed** [comment]
reviewed & signed off

**Pull request merged**